python/cudf_polars/cudf_polars/patch.py

```diff
@@ -0,0 +1,101 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import functools
+
+import polars as pl
+from polars.lazyframe.frame import InProcessQuery, wrap_df
+
+from cudf_polars.dataframe import DataFrame
+from cudf_polars.plan import execute_plan
+
+_WAS_PATCHED = False
+
+
+def patch_collect(use_gpu_default=False, cpu_fallback_default=True):
+    """Monkey patch LazyFrame.collect to enable GPU execution."""
+    global _WAS_PATCHED
+    if _WAS_PATCHED:
+        return
+    _WAS_PATCHED = True
+    import rmm
+
+    rmm.reinitialize(pool_allocator=True)
+
+    # TODO: docstring is not correct
+    @functools.wraps(pl.LazyFrame.collect)
+    def collect_with_gpu(
+        self,
+        *,
+        use_gpu: bool = use_gpu_default,
+        return_on_gpu: bool = False,
+        cpu_fallback: bool = cpu_fallback_default,
+        type_coercion: bool = True,
+        predicate_pushdown: bool = True,
+        projection_pushdown: bool = True,
+        simplify_expression: bool = True,
+        slice_pushdown: bool = True,
+        comm_subplan_elim: bool = True,
+        comm_subexpr_elim: bool = True,
+        no_optimization: bool = False,
+        streaming: bool = False,
+        background: bool = False,
+        _eager: bool = False,
+    ):
+        if streaming or _eager or background:
+            use_gpu = False
+        if no_optimization or _eager:
+            predicate_pushdown = False
+            projection_pushdown = False
+            slice_pushdown = False
+            comm_subplan_elim = False
+            comm_subexpr_elim = False
+
+        if streaming:
+            comm_subplan_elim = False
+        ldf = self._ldf.optimization_toggle(
+            type_coercion,
+            predicate_pushdown,
+            projection_pushdown,
+            simplify_expression,
+            slice_pushdown,
+            comm_subplan_elim,
+            comm_subexpr_elim,
+            streaming,
+            _eager,
+        )
+        if use_gpu:
+            # This can't fail
+            visitor = ldf.visit()
+            try:
+                gdf = execute_plan(visitor, profile=False)
```

Comment:
On the tip of this branch currently it seems `profile=` is unsupported, however tests pass with that kwarg removed.

Reply:
Can you please do that and push here?